### PR TITLE
Support downstream protocol in passthrough cluster

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/24706

### CODE REVIEW

File: tests/integration/telemetry/outboundtrafficpolicy/helper.go [hunk 1]
```diff
@@ -321,7 +325,7 @@ func setupEcho(t *testing.T, ctx resource.Context, mode TrafficPolicy) (echo.Ins
 		With(&dest, echo.Config{
 			Service:   "destination",
 			Namespace: appsNamespace,
-			Subsets:   []echo.SubsetConfig{{}},
+			Subsets:   []echo.SubsetConfig{{Annotations: echo.NewAnnotations().SetBool(echo.SidecarInject, false)}},
```
Comment (howardjohn) [inline tests/integration/telemetry/outboundtrafficpolicy/helper.go:328]: This is more accurate to simulating external traffic as we don't rely on envoy->envoy doing things that external services won't do

File: tests/integration/telemetry/outboundtrafficpolicy/traffic_allow_any_test.go [hunk 2]
```diff
@@ -45,17 +58,61 @@ func TestOutboundTrafficPolicy_AllowAny(t *testing.T) {
...
+				PromQueryFormat: `sum(istio_tcp_connections_opened_total{reporter="source",destination_service_name="PassthroughCluster"})`,
+				ResponseCode:    []string{"200"},
+				Metadata:        map[string]string{"Proto": "HTTP/2.0"},
 			},
 		},
 		{
 			Name:     "HTTP Traffic Egress",
 			PortName: "http",
 			Host:     "some-external-site.com",
-			Gateway:  true,
```
Comment (howardjohn) [inline tests/integration/telemetry/outboundtrafficpolicy/traffic_allow_any_test.go:54]: swapped this out with more flexible "Metadata" check

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 3]
```diff
@@ -240,6 +240,7 @@ func (cb *ClusterBuilder) buildDefaultPassthroughCluster() *cluster.Cluster {
 		ClusterDiscoveryType: &cluster.Cluster_Type{Type: cluster.Cluster_ORIGINAL_DST},
 		ConnectTimeout:       gogo.DurationToProtoDuration(cb.push.Mesh.ConnectTimeout),
 		LbPolicy:             cluster.Cluster_CLUSTER_PROVIDED,
+		ProtocolSelection:    cluster.Cluster_USE_DOWNSTREAM_PROTOCOL,
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:243]: This is the meat of the change, which fixes the actual bug. Everything else is testing

File: pkg/test/echo/cmd/client/main.go [hunk 4]
```diff
@@ -118,6 +119,8 @@ func init() {
 	rootCmd.PersistentFlags().StringVar(&caFile, "ca", "/cert.crt", "CA root cert file")
 	rootCmd.PersistentFlags().StringVar(&msg, "msg", "HelloWorld",
 		"message to send (for websockets)")
+	rootCmd.PersistentFlags().BoolVar(&http2, "http2", false,
```
Comment (howardjohn) [inline pkg/test/echo/cmd/client/main.go:122]: Currently we have no way for the client to send h2, so this enables that.

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go (hunk 3 above)
Comment (lambdai) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:243]: Great diagnose!

File: pkg/test/echo/cmd/client/main.go (hunk 4 above)
Comment (lambdai) [reply pkg/test/echo/cmd/client/main.go:122]: TIL there is explicit standard for "prior knowledge"

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go (hunk 3 above)
Comment (nrjpoddar) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:243]: This is very interesting and I'm surprised that it has gone unnoticed until now.
With the "Use_configured_protocol" do we set it to HTTP 1/1?

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:243]: Yes, lack of http2_protocol_options == use h1

Comment (mandarjog) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:243]: The reason for it going unnoticed is that people that use grpc do not go thru this path, the port are explicitly marked as grpc.

**Comment by mandarjog** [comment]
@howardjohn do you need to update API version ? there are compiler errors.

**Comment by howardjohn** [comment]
@mandarjog just fixed them, forgot to update part of the tests.

**Comment by nrjpoddar** [comment]
Add a release note for 1.7, should we fix it in earlier releases?

**Comment by howardjohn** [comment]
Yeah this probably warrants a backport. Should be trivial for the code change, not sure how hard the test changes will be to backport...

**Comment by skydoctor** [comment]
This is a great find and fix! Any plans on back-porting to 1.4?

**Comment by howardjohn** [comment]
No, see https://istio.io/latest/news/support/announcing-1.4-eol-final/

**Comment by KinglyWayne** [comment]
Really Great fix. Any plan to backport to 1.6.x branch?
As we are using istio 1.6.x, we met grpc passthrough downgrade to http1.1 problem (some grpc request direct sent to pod ip). Right now we have to add service entry for these service, however this is not really neccessary for us.

**Comment by HelloChenHZ** [comment]
Would this issue fixed in istio 1.6.x?

**Comment by HelloChenHZ** [comment]
> Really Great fix. Any plan to backport to 1.6.x branch?
> As we are using istio 1.6.x, we met grpc passthrough downgrade to http1.1 problem (some grpc request direct sent to pod ip). Right now we have to add service entry for these service, however this is not really neccessary for us.

Set the host of service name for the service entry? I tried but still failed.
********************************************
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: XXX
  namespace: lb
spec:
  hosts:
  - XXX.lb.svc.cluster.local
  ports:
  - number: 19898
    name: grpc-port
    protocol: TCP
  resolution: DNS

**Comment by HelloChenHZ** [comment]
> Really Great fix. Any plan to backport to 1.6.x branch?
> As we are using istio 1.6.x, we met grpc passthrough downgrade to http1.1 problem (some grpc request direct sent to pod ip). Right now we have to add service entry for these service, however this is not really neccessary for us.
How to set service entry for grpc service? For example, the service entry host is alway domain instead of grpc service name.

**Pull request merged**